### PR TITLE
[Sutton] Waste front end

### PR DESCRIPTION
**Pull request by dracos**
[skip changelog]
I reviewed the display stuff in the commit, so it's more the functionality etc that could do with a look over.

### CODE REVIEW

**Review by struan — APPROVED** [review]
Only comment is that it feels like there's a lot of shared code between this and Kingston and it might be good to share it for ease of maintenance if they are going to work the same?

Also, there are some differences (mostly in commented out code) and it's not clear if that's intentional and I'm not sure how to check.

File: perllib/FixMyStreet/Cobrand/Sutton.pm [hunk 1]
```diff
@@ -0,0 +1,815 @@
...
+sub garden_service_name { 'garden waste collection service' }
+sub garden_service_id { GARDEN_WASTE_SERVICE_ID }
+sub garden_current_subscription { shift->{c}->stash->{services}{+GARDEN_WASTE_SERVICE_ID} }
+sub get_current_garden_bins { shift->garden_current_subscription->{garden_bins} }
+
+sub garden_subscription_type_field { 'Request_Type' }
+sub garden_subscription_container_field { 'Subscription_Details_Containers' }
+sub garden_echo_container_name { 'SLWP - Containers' }
+
+sub garden_current_service_from_service_units {
```
[the remark this answers is not in the document]
Comment (dracos) [reply perllib/FixMyStreet/Cobrand/Sutton.pm:199]: Yeah, as you say we now see they share quite a bit.

File: perllib/FixMyStreet/Cobrand/Sutton.pm [hunk 2]
```diff
@@ -0,0 +1,815 @@
...
+#    return 1 if $event->{ResolutionCodeId} && $event->{ResolutionCodeId} != 584; # Out of Stock TODO Check this
+    return 0;
+}
+
+sub _parse_events {
+    my $self = shift;
+    my $events_data = shift;
+    my $events;
+    foreach (@$events_data) {
+        my $event_type = $_->{EventTypeId};
```
[the remark this answers is not in the document]
Comment (dracos) [reply perllib/FixMyStreet/Cobrand/Sutton.pm:440]: I imagine they will be the same in most respects - only commented out in Kingston as Kingston was garden bins only, and now we're doing Sutton everything, and then it'll be Kingston non-garden bins.

File: perllib/FixMyStreet/Cobrand/Sutton.pm [hunk 3]
```diff
@@ -0,0 +1,815 @@
...
+    my $c = $self->{c};
+
+    my $address = $c->stash->{property}->{address};
+    my $service = $c->stash->{services}{$id}{service_name};
+    $data->{title} = "Report missed $service";
+    $data->{detail} = "$data->{title}\n\n$address";
+    $c->set_param('service_id', $id);
+}
+
+#sub waste_munge_enquiry_data {
```
[the remark this answers is not in the document]
Comment (dracos) [reply perllib/FixMyStreet/Cobrand/Sutton.pm:619]: Yep

File: perllib/FixMyStreet/Cobrand/Sutton.pm [hunk 4]
```diff
@@ -0,0 +1,815 @@
...
+        2243 => "$base/large-communal-black", # Communal refuse
+        2248 => "$base/bin-brown", # Communal food
+        2249 => "$base/bin-grey-green-lid-recycling", # Communal paper
+        2250 => "$base/large-communal-green", # Communal recycling
+    };
+    return $images->{$service_id};
+}
+
+sub waste_staff_choose_payment_method { 1 }
+sub waste_cheque_payments { shift->{c}->stash->{staff_payments_allowed} }
```
[the remark this answers is not in the document]
Comment (dracos) [reply perllib/FixMyStreet/Cobrand/Sutton.pm:169]: Will ask

[the remark this answers is not in the document]
Comment (dracos) [reply perllib/FixMyStreet/Cobrand/Sutton.pm:169]: Yep, we assume in same way

**Pull request merged**